pkg/controllers/management/certsexpiration/certsexpiration.go

```diff
@@ -0,0 +1,84 @@
+package certsexpiration
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	"github.com/rancher/rancher/pkg/clustermanager"
+	"github.com/rancher/rancher/pkg/rkecerts"
+	rkeCluster "github.com/rancher/rke/cluster"
+	v1 "github.com/rancher/types/apis/core/v1"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// This controller handles cert expiration for local cluster only
+func Register(ctx context.Context, management *config.ManagementContext, clusterManager *clustermanager.Manager) {
+	c := &certsExpiration{
+		clusterClient:   management.Management.Clusters(""),
+		configMapLister: management.Core.ConfigMaps("kube-system").Controller().Lister(),
+	}
+	management.Management.Clusters("").AddHandler(ctx, "certificate-expiration", c.sync)
+}
+
+type certsExpiration struct {
+	clusterClient   v3.ClusterInterface
```

Comment:
Update to `clusters`, the naming conventions should be `<type>s` for the interface or `<type>Lister`